tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4528,14 +4523,55 @@ Status ConvertFusedBatchNorm(OpConverterParams* params) {
     }
   }
 
-  nvinfer1::ScaleMode mode = nvinfer1::ScaleMode::kCHANNEL;
-  nvinfer1::IScaleLayer* layer = params->converter->network()->addScale(
-      *tensor->trt_tensor(), mode, combined_offset_weights->GetTrtWeights(),
-      combined_scale_weights->GetTrtWeights(),
-      nvinfer1::Weights{nvinfer1::DataType::kFLOAT, nullptr, 0});
-  TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
-  params->converter->SetLayerName(layer, node_def);
-  ITensorProxyPtr output_tensor = layer->getOutput(0);
+  ITensorProxyPtr output_tensor;
+
+  if (data_format == "NCHW") {
+    nvinfer1::ScaleMode mode = nvinfer1::ScaleMode::kCHANNEL;
+    nvinfer1::IScaleLayer* layer = params->converter->network()->addScale(
+        *tensor->trt_tensor(), mode, combined_offset_weights->GetTrtWeights(),
+        combined_scale_weights->GetTrtWeights(),
+        nvinfer1::Weights{nvinfer1::DataType::kFLOAT, nullptr, 0});
+    TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
+    params->converter->SetLayerName(layer, node_def);
+    output_tensor = layer->getOutput(0);
+  }
+  if (data_format == "NHWC") {
```

Comment:
To handle this branch, couldn't we simply use [IScaleLayer::setChannelAxis()](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_scale_layer.html#a72e83b6fccead7347348b43cf81a3df2) (in TRT8)?

Reply:
That was my first attempt actually, but turns out Scale's channel axis doesn't work as you'd expect it. It just defines the "separator" dimension between non-spatial and spatial dimensions. So by setting channel axis = 3, it tries to compute the scaling only on the spatial dims > 3, for which there are none left anymore in NHWC layout and so it fails.